MSAL/test/unit/MSALDrsDiscoveryRequestTests.m

```diff
@@ -0,0 +1,156 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import "MSALTestCase.h"
+#import "MSALDrsDiscoveryRequest.h"
+#import "MSALTestURLSession.h"
+#import "MSALDrsDiscoveryResponse.h"
+
+@interface MSALDrsDiscoveryRequestTests : MSALTestCase
+
+@end
+
+@implementation MSALDrsDiscoveryRequestTests
+
+- (void)setUp
+{
+    [super setUp];
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+}
+
+- (void)tearDown
+{
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+    [super tearDown];
+}
+
+
+- (void)testUrlForDrsDiscoveryForDomain_whenNoDomain_shouldReturnNil
+{
+    XCTAssertNil([MSALDrsDiscoveryRequest urlForDrsDiscoveryForDomain:nil adfsType:MSAL_ADFS_CLOUD]);
+    XCTAssertNil([MSALDrsDiscoveryRequest urlForDrsDiscoveryForDomain:nil adfsType:MSAL_ADFS_ON_PREMS]);
+}
+
+- (void)testUrlForDrsDiscoveryForDomain_whenDomainAndOnPrems_shouldReturnUrl
+{
+    NSString *expectedUrlString = @"https://enterpriseregistration.somedomain.com/enrollmentserver/contract?api-version=1.0";
+    NSURL *resultUrl = [MSALDrsDiscoveryRequest urlForDrsDiscoveryForDomain:@"somedomain.com" adfsType:MSAL_ADFS_ON_PREMS];
+    XCTAssertNotNil(resultUrl);
+    XCTAssertEqualObjects(resultUrl.absoluteString, expectedUrlString);
+}
+
+- (void)testUrlForDrsDiscoveryForDomain_whenDomainAndOnCloud_shouldReturnUrl
+{
+    NSString *expectedUrlString = @"https://enterpriseregistration.windows.net/somedomain.com/enrollmentserver/contract?api-version=1.0";
+    NSURL *resultUrl = [MSALDrsDiscoveryRequest urlForDrsDiscoveryForDomain:@"somedomain.com" adfsType:MSAL_ADFS_CLOUD];
+    XCTAssertNotNil(resultUrl);
+    XCTAssertEqualObjects(resultUrl.absoluteString, expectedUrlString);
+}
+
+- (void)testQueryEnrollmentServerEndpointForDomain_whenDomainNil_shouldReturnError
+{
+    [MSALDrsDiscoveryRequest queryEnrollmentServerEndpointForDomain:nil
+                                                           adfsType:MSAL_ADFS_CLOUD
+                                                            context:nil
+                                                    completionBlock:^(MSALDrsDiscoveryResponse *response, NSError *error)
+     {
+         XCTAssertNil(response);
+         XCTAssertNotNil(error);
+         
+         XCTAssertTrue(error.code == MSALErrorInvalidParameter);
+         XCTAssertTrue([error.userInfo[MSALErrorDescriptionKey] containsString:@"Domain"]);
```

Comment:
Add a dispatch_semaphore_signal/_wait here, just in case someone ever goes along and adds a dispatch async in this code path.